SimCalorimetry/EcalEBTrigPrimProducers/plugins/EcalEBTrigPrimPhase2ESProducer.cc

```diff
@@ -0,0 +1,694 @@
+              std::string result5 = oss.str();
+
+              st1.append(result5);
+              std::cout << "" << st1 << std::endl;
+            }
+          }
+        }
+      }
+
+      mapStrip_[0][id] = param;
+    }
+
+    if (dataCard == "STRIP_EE") {
+      gis >> std::dec >> id;
+
+      std::string st6;
+
+      if (flagPrint_) {
+        std::cout << dataCard << " " << std::dec << id << std::endl;
+      }
+
+      param.clear();
+      for (int i = 0; i < NBstripparams[1]; i++) {
+        gis >> std::hex >> data;
+        param.push_back(data);
+
+        if (flagPrint_) {
+          if (i == 0) {
+            std::cout << "0x" << std::hex << data << std::endl;
+          } else if (i == 1) {
+            std::cout << " " << std::hex << data << std::endl;
+          } else if (i > 1) {
+            std::ostringstream oss;
+            if (i == 2) {
+              oss << "0x" << std::hex << data;
+              std::string result4 = oss.str();
+              st6.append(result4);
+            } else if (i == 3) {
+              std::ostringstream oss;
+              oss << " 0x" << std::hex << data;
+              std::string result5 = oss.str();
+
+              st6.append(result5);
+              std::cout << "" << st6 << std::endl;
+            }
+          }
+        }
+      }
+
+      mapStrip_[1][id] = param;
+    }
+
+    if (dataCard == "TOWER_EE") {
+      gis >> std::dec >> id;
+
+      if (flagPrint_)
+        std::cout << dataCard << " " << std::dec << id << std::endl;
+
+      param.clear();
+      for (int i = 0; i < 2; i++) {
+        gis >> std::hex >> data;
+        param.push_back(data);
+
+        if (flagPrint_) {
+          if (i == 1) {
+            std::cout << "0x" << std::dec << data << std::endl;
+          } else {
+            std::cout << " " << std::dec << data << std::endl;
+          }
+        }
+      }
+
+      // std::cout<<std::endl ;
+      mapTower_[1][id] = param;
+    }
+
+    if (dataCard == "TOWER_EB") {
+      gis >> std::dec >> id;
+
+      if (flagPrint_)
+        std::cout << dataCard << " " << std::dec << id << std::endl;
+
+      param.clear();
+      for (int i = 0; i < 3; i++) {
+        gis >> std::dec >> data;
+
+        if (flagPrint_) {
+          std::cout << " " << std::dec << data << std::endl;
+        }
+
+        param.push_back(data);
+      }
+
+      // std::cout<<std::endl ;
+      mapTower_[0][id] = param;
+    }
+  }
+}
+
+std::vector<int> EcalEBTrigPrimPhase2ESProducer::getRange(
+    int subdet, int tccNb, int towerNbInTcc, int stripNbInTower, int xtalNbInStrip) {
+  std::vector<int> range;
+  if (subdet == 0) {
+    // Barrel
+    range.push_back(37);  // stccNbMin
+    range.push_back(73);  // tccNbMax
+    range.push_back(1);   // towerNbMin
+    range.push_back(69);  // towerNbMax
+    range.push_back(1);   // stripNbMin
+    range.push_back(6);   // stripNbMax
+    range.push_back(1);   // xtalNbMin
+    range.push_back(6);   // xtalNbMax
+  } else {
```

Comment:
Why is the endcap needed for Phase 2?

Reply:
The method getRange is from Phase I.  It is not used anywhere, but is a reminder that something alike will need to be implemented once we get the mapping of EB to BCPs. I remove the else referring to the endcap